public/app/features/dashboard/components/SaveDashboard/types.ts

```diff
@@ -27,6 +29,16 @@ export interface SaveDashboardAsOptions {
   description?: string;
 }
 
+export interface SaveDashboardCommandV2 {
+  dashboard: DashboardV2Spec;
+  message?: string;
+  folderUid?: string;
+  overwrite?: boolean;
+  showErrorAlert?: boolean;
+
+  // When loading dashboards from k8s, we need to have access to the metadata wrapper
+  k8s?: Partial<ObjectMeta>;
```

Comment:
Don't understand why for the v2 we need to use this hack? It was introduce to allow legacy and v0 to use the same command really. For v2 we shouldn't do the same thing. If we are introducing the new command we should provide the metadata explicitly when needed, or provide the full k8s resource as the dashboard to save.